# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmInstance.cpp
# Commit: 40f861f62a0f
# Full Hash: 40f861f62a0f58040cb6cadf073bc5e97a44231d
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2021-12-23 09:34:36
# Regressor Bug: 1642412
# File Overlap Count: 7
# Description:
#   Bug 1642412 - wasm: Optimize baseline table.size/get/set. r=lth
#   
#   This commit adds a fast path for table.get/set when the table
#   representation is anyref, and an unconditional fast path for
#   table.length.
# ==============================================================================

diff -r 58a45475b524 -r 40f861f62a0f js/src/wasm/WasmInstance.cpp
--- a/js/src/wasm/WasmInstance.cpp	Wed Dec 22 19:09:54 2021 +0200
+++ b/js/src/wasm/WasmInstance.cpp	Wed Dec 22 17:36:29 2021 +0000
@@ -1732,7 +1732,7 @@
     const TableDesc& td = metadata().tables[i];
     TableTls& table = tableTls(td);
     table.length = tables_[i]->length();
-    table.functionBase = tables_[i]->functionBase();
+    table.elements = tables_[i]->tlsElements();
   }
 
   // Add observer if our memory base may grow
@@ -2601,7 +2601,7 @@
     if (tables_[i] == theTable) {
       TableTls& table = tableTls(metadata().tables[i]);
       table.length = tables_[i]->length();
-      table.functionBase = tables_[i]->functionBase();
+      table.elements = tables_[i]->tlsElements();
     }
   }
 }